### ISSUE TITLE
Vaults not Combining ItemStackHandlers

### ISSUE THREAD

**Issue by Talia-12** [Fabricators-of-Create/Create#448]
### Describe the Bug

Attempting to use item vaults and found that when extracting, chutes & funnels were unable to extract more than 20 stacks, even though upon breaking the vaults almost all (bar presumably the controller) still contained items. Comparators and Stockpile Switches both report the item vault as empty even when it is not.

Works properly when assembled into a minecart contraption (in that portable storage interfaces can be used to extract from all).

### Reproduction Steps

I haven't been able to consistently reproduce it yet. I removed the vault completely and rebuilt it and the issue was still present, but building a new vault in a different location didn't reproduce. I'd been using the vaults in minecart contraptions, and using another mod to teleport the minecart they were attached to - both could plausibly have helped result in the error.

### Expected Result

the vaults contain 20 stacks each, and I had 45, so I expected to extract 900 stacks rather than 20.

### Operating System

Windows 10

### Mod Version

0.4.1

### Minecraft Version

1.18.2

### Other Mods

advancementfix-0.1.0
allium-0.2.2+1.18.2
architectury-4.5.76-fabric
badpackets-fabric-0.1.2
Botania-1.18.2-434-FABRIC
cardinal-components-api-4.1.4
carpet-extra-1.18.2-1.4.72
cc-restitched-1.100.8
chalk-2.0.1+1.18.2
chipped-1.2
cloth-config-6.2.62-fabric
Controlling-fabric-1.18.2-9.0+21
crawl-0.10.6
create-mc1.18.2_v0.4.1+637
DarkPaintings-Forge-1.18.2-10.0.3
ears-fabric-1.17-1.4.5_01
expanded-trident-enchanting-1.0.3
ExtendedDrawers-1.3.1+mc.1.18.2
extremesoundmuffler-3.27_fabric-1.18.x
fabric-carpet-1.18.2-1.4.69+v220331
fabric-language-kotlin-1.8.1+kotlin.1.7.0
farmers-delight-fabric-0.2.1
ferritecore-4.2.1-fabric
glassential-fabric-1.3.0
graphlib-0.2.4+1.18.2
half-doors-1.18.2-fabric-1.3.7
hexcasting-fabric-1.18.2-0.9.1
Incorporeal-3-fabric-1.18.2-4
indium-1.0.6+mc1.18.2
journeymap-1.18.2-5.8.5-fabric
kubejs-fabric-1802.5.4-build.516
lanterns-belong-on-walls-1.3.0-1.18.x
lazydfu-0.1.2
litematica-fabric-1.18.2-0.11.3
lithium-fabric-mc1.18.2-0.7.10
malilib-fabric-1.18.2-0.12.1
minihud-fabric-1.18.2-0.22.0
modmenu-3.2.3
MouseTweaks-fabric-mc1.18-2.22
notenoughcrashes-4.1.6+1.18.2-fabric
Patchouli-1.18.2-71.1-FABRIC
paucal-fabric-1.18.2-0.4.7
Paxi-1.18.2-Fabric-2.0
Pehkui-3.3.3+1.14.4-1.19.1
phosphor-fabric-mc1.18.x-0.8.1
qsl-1.1.0-beta.19_qfapi-1.0.0-beta.22_fapi-0.56.0_mc-1.18.2
rhino-fabric-1802.1.13-build.179
RoughlyEnoughItems-8.3.510
skiptransitions-1.3.2+1.18.x
sodium-fabric-mc1.18.2-0.4.1+build.15
tooltipfix-1.0.4-1.18
travelers-backpack-1.18.2-7.1.3
traverse-5.1.0
trinkets-3.3.1
tweakeroo-fabric-1.18.2-0.13.3
twigs-fabric-1.1.3-patch2+1.18.2
veinmining-fabric-0.0.13-1.18.2
wthit-fabric-4.11.0
youre-in-grave-danger-1.3.7
YungsApi-1.18.2-Fabric-2.0.8
YungsBetterDesertTemples-1.18.2-Fabric-1.2
Zoomify-1.6.0

### Additional Context

Also potentially of note is that this bug occurred when using Quilt, rather than Fabric. I have yet to try and reproduce it in Fabric.

**Comment by TropheusJ** [comment]
I can't reproduce either, let me know if a way is found.

**Comment by Talia-12** [comment]
okay, I successfully reproduced it with a new vault! I've got two pictures of the setup below, but basically it's a vault with glue sticking it together, a comparator reading its storage fill level, with the output being sent to a cart assembler with a controller rail underneath it.
![image](https://user-images.githubusercontent.com/17948170/180201962-5f28639d-2bd7-4614-a9c5-08cdcecba96d.png)
![image](https://user-images.githubusercontent.com/17948170/180201979-54e4ab4f-5185-47a1-baa0-3fe10e39ec08.png)

After more testing it seems like the critical thing is that the vault form while it has a comparator or stockpile switch reading from it. Content observers don't cause the bug.

(Of note is this time I did it in a Fabric instance, rather than with Quilt)
